client/dashboard/domains/domain-transfer/index.tsx

```diff
@@ -0,0 +1,289 @@
+import { domainLockMutation, domainTransferCodeMutation } from '../../app/queries/domain-transfer';
+import { domainRoute } from '../../app/router/domains';
+import { ActionList } from '../../components/action-list';
+import InlineSupportLink from '../../components/inline-support-link';
+import Notice from '../../components/notice';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import RouterLinkButton from '../../components/router-link-button';
+import { SectionHeader } from '../../components/section-header';
+import { DomainSubtype } from '../../data/domains';
+import { formatDate } from '../../utils/datetime';
+import SelectIpsTag from './select-ips-tag';
+
+const RESTRICTED_TRANSFER_TLDS = [ 'uk', 'fr', 'ca', 'de', 'jp' ];
+const TRANSFER_LOCK_GRACE_PERIOD_MS = 60 * 24 * 60 * 60 * 1000; // 60 days
+
+export function getTopLevelOfTld( domainName: string ): string {
+	return domainName.substring( domainName.lastIndexOf( '.' ) + 1 );
+}
+
+export default function DomainTransfer() {
+	const { domainName } = domainRoute.useParams();
+	const { data: domain } = useSuspenseQuery( domainQuery( domainName ) );
+	const locale = useLocale();
+	const updateDomainLockMutation = useMutation( domainLockMutation( domainName ) );
+	const requestTransferCodeMutation = useMutation( domainTransferCodeMutation( domainName ) );
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const isUpdatingDomainLock = updateDomainLockMutation.isPending;
+	const isRequestingTransferCode = requestTransferCodeMutation.isPending;
+
+	const handleToggleChange = ( enabled: boolean ) => {
+		updateDomainLockMutation.mutate( enabled, {
+			onSuccess: () => {
+				createSuccessNotice( __( 'Transfer lock setting saved.' ), { type: 'snackbar' } );
+			},
+			onError: () => {
+				createErrorNotice( __( 'Failed to save transfer lock settings.' ), {
+					type: 'snackbar',
+				} );
+			},
+		} );
+	};
+
+	const renderTransferInfo = () => {
+		return (
+			<Notice title={ __( 'How do transfers work?' ) }>
+				{ createInterpolateElement(
+					__(
+						'Transferring a domain within WordPress.com is immediate. However, transferring a domain to another provider can take 5–7 days during which no changes to the domain can be made. Read <link>Transfer a domain to another registrar</link> before starting a transfer.'
+					),
+					{
+						link: <InlineSupportLink supportContext="transfer-domain-to-another-registrar" />,
+					}
+				) }
+			</Notice>
+		);
+	};
+
+	const renderInternalTransferOptions = () => {
+		const actions = [];
+		if ( ! domain.is_domain_only_site && domain.can_transfer_to_any_user ) {
+			const description =
+				domain.subtype.id === DomainSubtype.DOMAIN_CONNECTION
+					? __( 'Transfer this domain connection to any administrator on this site' )
+					: __( 'Transfer this domain to any administrator on this site' );
```

Comment:
```suggestion
					? __( 'Transfer this domain connection to any administrator on this site.' )
					: __( 'Transfer this domain to any administrator on this site.' );
```

Sames goes for other description text in this PR.